Add client info screen

jz1210 opened the pull request:
The input part is almost done. Next plan is give auto complete function for ListView and ComboBox.

jz1210 committed:
MoveSqlCommand

Move one insert command to dbconnection.

jz1210 committed:
Move sql command

Moved command to dbconnection class.

jz1210 committed:
Start on new screen

Create add client inof screen with controller

jz1210 committed:
Update AddClientInfoScreen.fxml

UI demo

jz1210 committed:
Set up

Set up the function id for elements and put the name in controller

jz1210 committed:
Update AddClientInfoScreenController.java

Commit:
Fix and set up object

Set up object for 10 cards info, and fixed some small error.

jz1210 committed:
Delete the space for comment line

Delete the space for comment line and add some comment

Commit:
Pass name to add client info

Add a label that could change name based on the select client and pass that to add client info

Commit:
Show the basic info from database

Client Info screen will show the basic info from database

Commit:
Set up the alert screen

Set up the alert screen with the data in database based on client ID

Commit:
Set up each tab

Show the data from database on every tab

Commit:
Insert and update command

Insert and update command with add client info screen

Commit:
Demo for add client info

nameless-Q committed:
ListView

Use list view to show more than one data in add client info

nameless-Q committed:
Refresh when save new things for provider tab

nameless-Q committed:
update provider list

update listview into provider

nameless-Q committed:
Make every tab work

Most tab are works and I need give more detail comments.

nameless-Q committed:
Make diagnose editable to update

nameless-Q committed:
Add edit button

babbett commented on AMPM-Guardian/src/ampm/Alert.java:
```diff
@@ -0,0 +1,100 @@
+package ampm;
+
+/**
+ * This one is for alert in add client info screen
```
These comments are a little bit confusing. While it is used in the client info screen, this probably isn't the only place where it will be used. Same with all the other classes you created.

jz1210 replied:
I see

babbett commented on AMPM-Guardian/src/ampm/AddClientInfoScreenController.java:
```diff
@@ -0,0 +1,960 @@
+                try {
+                    setUpMonitorList();
+                    //Show the releated information when click one item in ListView
+                    String query = "SELECT * FROM AMPM.Diagnose WHERE ClientID ='" + this.clientID + "'AND Diagnosis = '"
```
I really want us to stay away from creating sql queries in the controller classes. We have the DBmanager class for this. You should move the queries in this file to that class and set up the logic like that. It will make transitioning to the correct online/offline portion much easier. Unless you can come up with a reason proving me wrong.

jz1210 replied:
Yeah, I planed to move them into DBmanager class and I forgot.

nameless-Q committed:
Move sql queries

Move sql queries into the controller classes

nameless-Q committed:
Add new button

Remove the add in the ListView and change edit button to new button

babbett committed:
Fixed some small bugs

babbett committed:
Merge branch 'add-client-info-screen' of https://github.com/gordon-cs/ampm-catalyst into add-client-info-screen

nameless-Q committed:
Fix the save button

nameless-Q committed:
Fix some error and some UI problem

Need continue working on UI problem and need figure out the layout for the family history and monitor

nameless-Q committed:
Make UI layout better

nameless-Q committed:
Fix some typo

nameless-Q committed:
Fix some typo

nameless-Q committed:
Add two screen

This two screen is use to add diagnoses and providers

nameless-Q committed:
Screen flow

Click new client will not close home screen

nameless-Q committed:
ComboBoxManage Class

Get a demo about ComboBoxManage  screen

nameless-Q committed:
Change method name

nameless-Q committed:
Change the new client input info

Add DOB in add client  screen and change client ID into unique identifier.

nameless-Q committed:
Fix some error and put pdfbox in this branch

Fix client class after use unique identifier. Put PDFBox libraries and PDFPrint class in this branch.

nameless-Q committed:
Update project.properties

nameless-Q committed:
Added other 3 Print Card method

Add 3 print card button and print  method

nameless-Q committed:
Add textfield for client ID

Have a textfield  in addClientInfo Screen to hold the client ID(unique identifer)

nameless-Q committed:
Fix the provider set up

nameless-Q committed:
Auto Update client list view

Change the location for output and client ListView can auto refresh but have small error.

nameless-Q committed:
Fix some errors

nameless-Q committed:
Change the way to read name

nameless-Q committed:
Print extra page

Print extra page when the page was filled.

nameless-Q committed:
Update PDFPrint.java

All 4 top priority tabs could print extra page when one page was filled.

nameless-Q committed:
Change the way to pass data to AddClientInfo Screen

Change the way to pass data to AddClientInfo Screen. Make the arraylist store the client object instead of string

nameless-Q committed:
Fix some error and refresh the home screen

nameless-Q committed:
Alert object class rename to alerts

Change alert object class name to alerts, because there are some specific alert class for information alert.

nameless-Q committed:
Alert message for print

Alert information for print and refresh listview(not work yet)

nameless-Q committed:
Set up provider type combo box

Connect provider type combo box to database

nameless-Q committed:
update ComboBox

update ComboBox in medication tab

nameless-Q committed:
Update medication comboBox

Fix the error about setValue will also point to setOnAction

nameless-Q committed:
Update

Update some combo box and rename the table in database

nameless-Q committed:
Update medication tab

Change one variable name and fix the error about setOnAction event

nameless-Q committed:
Insert medication

Insert medication to medication list if medication_list table don't have that

nameless-Q committed:
Update diagnoseTab

diagnose could could save with only diagnose information

nameless-Q committed:
Update some comment and delete some useless class

nameless-Q committed:
Delete redundant sclass
